spec/nexmo-jwt/jwt_builder_spec.rb

```diff
@@ -86,6 +86,35 @@
     end
   end
 
+  context 'with exp parameter provided in custom configuration block' do
+    before :each do
+      @builder = Nexmo::JWTBuilder.new(
+        application_id: '123456789',
+        private_key: './spec/nexmo-jwt/private_key.txt',
+        exp: 600
+      )
+    end
+
+    it 'includes the custom defined exp parameter in the builder' do
+      expect(@builder.exp).to eql(600)
+    end
+  end
+
+  context 'without exp parameter provided in configuration block' do
+    before :each do
+      @builder = Nexmo::JWTBuilder.new(application_id: '123456789', private_key: './spec/nexmo-jwt/private_key.txt')
+    end
+
+    it 'sets exp parameter to equal nil' do
+      expect(@builder.exp).to eql(nil)
+    end
+
+    it 'sets exp parameter in JWT class to equal the value of the ttl and iat parameters added together' do
+      hash = @builder.jwt.to_payload
+      expect(hash[:exp]).to eql(Time.now.to_i + 900)
```

Comment:
This can rail as a race condition unless `Time.now` is mocked

Reply:
Fixed